File: src/main/java/com/vc/easy/L266.scala
package com.vc.easy

object L266 {
  def canPermutePalindrome(s: String): Boolean = {
    import scala.collection.mutable
    val set = new mutable.HashSet[Char]()
    s.foreach(ch => {
      if(set.contains(ch)) set -= ch
      else set += ch
    })
    set.isEmpty || set.size == 1
  }
}

File: src/main/java/com/vc/easy/L266.java
package com.vc.easy;

import java.util.*;

class L266 {
    public boolean canPermutePalindrome(String s) {
        HashMap<Character, Integer> map = new HashMap<>();
        for(int i = 0; i < s.length(); i++) {
            map.put(s.charAt(i), map.getOrDefault(s.charAt(i), 0) + 1);
        }

        boolean foundOne = false;
        for(Map.Entry<Character, Integer> entry: map.entrySet()) {
            if(entry.getValue() % 2 != 0) {
                if(foundOne) return false;
                else foundOne = true;
            }
        }
        return true;
    }
}
